spacy_llm/tasks/summarization.py

```diff
@@ -0,0 +1,145 @@
+        template=template, examples=span_examples, max_n_words=max_n_words, field=field
+    )
+
+
+class SummarizationTask(SerializableTask[SummarizationExample]):
+    def __init__(
+        self,
+        template: str = _DEFAULT_SUMMARIZATION_TEMPLATE_V1,
+        examples: Optional[List[SummarizationExample]] = None,
+        max_n_words: Optional[int] = None,
+        field: str = "summary",
+    ):
+        """Default summarization task.
+
+        template (str): Prompt template passed to the model.
+        examples (Optional[Callable[[], Iterable[Any]]]): Optional callable that
+            reads a file containing task examples for few-shot learning. If None is
+            passed, then zero-shot learning will be used.
+        max_n_words (int): Max. number of words to use in summary.
+        field (str): The name of the doc extension in which to store the summary.
+        """
+        self._template = template
+        self._examples = examples
+        self._max_n_words = max_n_words
+        self._field = field
+        self._prompt_examples = examples or []
+        self._are_example_summaries_checked = False
+        if not Doc.has_extension(field):
+            Doc.set_extension(field, default=None)
+
+    def initialize(
+        self,
+        get_examples: Callable[[], Iterable["Example"]],
+        nlp: Language,
+        n_prompt_examples: int = 0,
+        **kwargs: Any,
+    ) -> None:
+        """Initializes prompt examples from Doc examples.
+        get_examples (Callable[[], Iterable["Example"]]): Callable that provides examples
+            for initialization.
+        nlp (Language): Language instance.
+        n_prompt_examples (int): How many prompt examples to infer from the provided Example objects.
+            0 by default. Takes all examples if set to -1.
+        """
+        for eg in get_examples():
+            if n_prompt_examples < 0 or len(self._prompt_examples) < n_prompt_examples:
+                self._prompt_examples.append(
+                    SummarizationExample(
+                        text=eg.reference.text,
+                        summary=getattr(eg.reference._, self._field),
+                    )
+                )
+
+    def _check_prompt_example_summary_len(self) -> None:
+        """Checks whether summaries of prompt examples are of expected lengths. Warns if they aren't."""
+        if self._max_n_words is None:
+            return
+
+        for pr_ex in self._prompt_examples:
+            len_summary = len(pr_ex.summary.split())
+            len_text = len(pr_ex.text.split())
+            if len_summary >= len_text:
+                warnings.warn(
+                    f"The provided example '{pr_ex.text[:30]}...' has a summary of length {len_summary} and a text "
+                    f"of length {len_text}. Ensure that your examples' summaries are shorter than their original "
+                    f"texts."
+                )
+            if len_summary > self._max_n_words:
```

Comment:
Given that our tokenization is crude, let's add some buffer here so we only warn when the summary is significantly larger? (like times 1.5 or 1.2 orso)

Reply:
Done in https://github.com/explosion/spacy-llm/pull/181/commits/e7b287a3ff35251b323d6e9566e2e5fcdb0f3977.